commit 009669cfad50beb10b3f63434863de231c6b408e
Author: Doug Ledford <dledford@redhat.com>
Date:   Tue May 15 14:19:19 2018 -0400

    RDMA/hfi1: Fix build error with debugfs disabled
    
    A recent patch set to rework the usage of debugfs and to add fault
    injection capabilities via debugfs files to the hfi1 driver introduced a
    build error that only shows up when debugfs is fully disabled.  The
    patchset mistakenly defines some empty stub functions in two different
    headers when debugfs is disabled.  Remove the set that shouldn't have
    been there to resolve the issue.
    
    Fixes: a74d5307caba ("IB/hfi1: Rework fault injection machinery")
    Signed-off-by: Doug Ledford <dledford@redhat.com>

diff --git a/drivers/infiniband/hw/hfi1/debugfs.h b/drivers/infiniband/hw/hfi1/debugfs.h
index 1c91461b108f..d5d824459fcc 100644
--- a/drivers/infiniband/hw/hfi1/debugfs.h
+++ b/drivers/infiniband/hw/hfi1/debugfs.h
@@ -118,21 +118,6 @@ static inline void hfi1_dbg_init(void)
 static inline void hfi1_dbg_exit(void)
 {
 }
-
-static inline bool hfi1_dbg_should_fault_rx(struct hfi1_packet *packet)
-{
-	return false;
-}
-
-static inline bool hfi1_dbg_should_fault_tx(struct rvt_qp *qp, u32 opcode)
-{
-	return false;
-}
-
-static inline bool hfi1_dbg_fault_suppress_err(struct hfi1_ibdev *ibd)
-{
-	return false;
-}
 #endif
 
 #endif                          /* _HFI1_DEBUGFS_H */